crates/services/p2p/src/service.rs

```diff
@@ -878,6 +877,7 @@ where
     B: Broadcast + 'static,
     T: TxPool + 'static,
 {
+    #[allow(clippy::arithmetic_side_effects)]
```

Comment:
Why do we need it?

Reply:
I guess clippy was ok when the future was boxed, but it started throwing this error when I removed it.

we still need the dependency because Runnable service uses it, I was thinking to remove in a follow up pr

Reply:
Maybe we can resolve it in this PR? Or is it too many changes?

Reply:
addressed in ff12199447

Reply:
i wonder why it was not caught previously? wouldn't a boxed future also still have the same property that overflow is possible? weird